libs/text-splitters/tests/unit_tests/test_text_splitters.py

```diff
@@ -2039,49 +2039,476 @@ def test_haskell_code_splitter() -> None:
     assert chunks == expected_chunks
 
 
-@pytest.mark.requires("lxml")
-def test_html_header_text_splitter(tmp_path: Path) -> None:
-    splitter = HTMLHeaderTextSplitter(
-        headers_to_split_on=[("h1", "Header 1"), ("h2", "Header 2")]
-    )
+@pytest.fixture
+@pytest.mark.requires("bs4")
+def html_header_splitter_splitter_factory() -> (
+    Callable[[List[Tuple[str, str]]], HTMLHeaderTextSplitter]
+):
+    """
+    Fixture to create an HTMLHeaderTextSplitter instance with given headers.
+    This factory allows dynamic creation of splitters with different headers.
+    """
 
-    content = """
-<h1>Sample Document</h1>
-    <h2>Section</h2>
-        <p id="1234">Reference content.</p>
+    def _create_splitter(
+        headers_to_split_on: List[Tuple[str, str]],
+    ) -> HTMLHeaderTextSplitter:
+        return HTMLHeaderTextSplitter(headers_to_split_on=headers_to_split_on)
 
-    <h2>Lists</h2>
-        <ul>
-            <li>Item 1</li>
-            <li>Item 2</li>
-            <li>Item 3</li>
-        </ul>
+    return _create_splitter
 
-        <h3>A block</h3>
-            <div class="amazing">
-                <p>Some text</p>
-                <p>Some more text</p>
-            </div>
+
+@pytest.mark.parametrize(
+    "headers_to_split_on, html_input, expected_documents, test_case",
+    [
+        (
+            # Test Case 1: Split on h1 and h2
+            [("h1", "Header 1"), ("h2", "Header 2")],
+            """
+            <html>
+                <body>
+                    <h1>Introduction</h1>
+                    <p>This is the introduction.</p>
+                    <h2>Background</h2>
+                    <p>Background information.</p>
+                    <h1>Conclusion</h1>
+                    <p>Final thoughts.</p>
+                </body>
+            </html>
+            """,
+            [
+                Document(
+                    page_content="Introduction", metadata={"Header 1": "Introduction"}
```

Comment:
> Headers: Each specified header (e.g., <h1>, <h2>) becomes a separate Document containing the header text.

This is different from the behavior before. Which situation is this useful for? Is this for a case where an HTML document only contains the <h1> header and no other content inside the header?

----

How would this be used downstream?

Should we include a boolean flag to toggle this behavior?

Reply:
When I was checking old behaviour. Sometimes it create a document contains only the header text and then the next one contains the whats in between. So, I wrote it to be consistently creating a document contains the header text and then next one contains the the content inside. There is no strong reason for doing it like that. So, I can change it.